scripts/calculate_epiweek.py

```diff
@@ -11,26 +13,22 @@
         formatter_class=argparse.ArgumentDefaultsHelpFormatter,
     )
     parser.add_argument("--metadata", required=True, help="metadata with a 'date' column")
+    parser.add_argument("--metadata-id-columns", default=["strain", "name", "Virus name"], nargs="+", help="names of valid metadata columns containing identifier information like 'strain' or 'name'")
     parser.add_argument("--attribute-name", default="epiweek", help="name to store annotations of epiweeks in JSON output")
     parser.add_argument("--output-node-data", required=True, help="node data JSON with epiweek annotations")
 
     args = parser.parse_args()
 
-    # Read metadata with pandas because Augur's read_metadata utility does not
-    # support metadata without a "strain" or "name" field.
-    metadata = pd.read_csv(
+    metadata = read_metadata(
         args.metadata,
-        sep=None,
-        engine="python",
-        skipinitialspace=True,
-        dtype={
-            "strain": "string",
-            "name": "string",
-        }
-    ).fillna("")
+        id_columns=args.metadata_id_columns,
+    )
 
-    # Find records with unambiguous dates.
-    metadata_with_dates = metadata.loc[~metadata["date"].str.contains("X"), ["strain", "date"]].copy()
+    # Find records with unambiguous dates. These must be complete date-like
+    # records in YYYY-MM-DD format.
+    date_pattern = re.compile(r"\d{4}-\d{2}-\d{2}")
```

Comment:
Add match for start/end, otherwise something like `2020-01-011` passes when it shouldn't.

```suggestion
    date_pattern = re.compile(r"^\d{4}-\d{2}-\d{2}$")
```

Reply:
Nice catch, @victorlin! I knew I was searching the start of the string with `re.match`, but I forgot about the end.